tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/AppFilterSuite.scala

```diff
@@ -302,6 +302,36 @@ class AppFilterSuite extends FunSuite {
       "3w", 2)
   }
 
+  test("full app name exact and 2-oldest-overall no match from filename") {
+    testFileSystemTimeAndStartAndAppFull(appsFullWithFsToTest, "2-oldest-overall", "nds", "nomatch",
+      "3w", 0)
+  }
+
+  test("full app name exact and 2-oldest-overall") {
+    testFileSystemTimeAndStartAndAppFull(appsFullWithFsToTest, "2-oldest-overall", "nds", "nds",
+      "3w", 2)
+  }
+
+  test("full app name exact and 2-newest-overall") {
+    testFileSystemTimeAndStartAndAppFull(appsFullWithFsToTest, "2-newest-overall", "ndsweeks2",
+      "nds", "3w", 1)
+  }
+
+  test("full app name exact and 1-newest-per-app-name") {
+    testFileSystemTimeAndStartAndAppFull(appsFullWithFsToTest, "1-newest-per-app-name", "nds",
+      "nds", "3w", 3)
```

Comment:
we don't validate we got the newest one

Reply:
overall I'd like to see more specific tests for this to make sure we are getting oldest and newest, also should have some without other filters in it to make sure it works standalone.

Reply:
I have added the tests without other filters. PTAL.